actions/defend_rush_buildings.py

```diff
@@ -0,0 +1,53 @@
+from sc2.constants import BUNKER, DRONE, PHOTONCANNON, PLANETARYFORTRESS, PROBE, SCV, SPINECRAWLER
+
+
+class DefendRushBuildings:
+    def __init__(self, ai):
+        self.ai = ai
+        self.rush_buildings = None
+
+    async def should_handle(self, iteration):
+        self.rush_buildings = self.ai.known_enemy_structures.not_flying.filter(
+            lambda building: any([building.distance_to(our_building) <= 30 for our_building in self.ai.units.structure])
+        )
+        return bool(self.rush_buildings)
+
+    def is_being_attacked(self, unit):
+        # only for enemy units, returns how often they are attacked
+        attackers = 0
+        near_units = self.ai.units.filter(lambda x: x.is_attacking)
+        for attacker in near_units:
+            if attacker.order_target == unit.tag:
+                attackers += 1
+        return attackers
+
+    async def handle(self, iteration):
+        # self.rush_buildings = self.ai.known_enemy_structures.closer_than(20, self.bases.first)
+        if self.rush_buildings and self.ai.bases:
```

Comment:
these conditions could be placed in the should handle

Reply:
self.rush_buildings is in there, already, so we dont need it in handle. ill fix that